salt/client/ssh/__init__.py

```diff
@@ -582,11 +596,40 @@ def handle_ssh(self, mine=False):
                 log.error("No matching targets found in roster.")
                 break
             if len(running) < self.opts.get("ssh_max_procs", 25) and not init:
-                try:
-                    host = next(target_iter)
-                except StopIteration:
+                if targets_queue:
+                    host = targets_queue.pop(0)
+                else:
                     init = True
                     continue
+                with salt.utils.files.flopen(self.session_flock_file, "w"):
```

Comment:
What would happen here is the lock file exists already? Isn't this raising an exception?

Reply:
It doesn't matter if the file exists already, after implementing the change file exists almost olways. Only posibility to call `flock` inside `salt.utils.files.flopen` makes sense. `salt.utils.files.flopen` is used as it's context aware.